File: test_dense.jl
# test_dense.jl
# Make a big dense matrix
require("julp.jl")

using Julp

function doTest()
  srand(10)
  N = 10000
  M = 1000
  A = randi(    10,M,N)
  b = N+randi(N*10,M)
  c = randi(    10,N)

  tic()
  m = Model("max")
  v = [ Variable(m,0,1,0,"x$j") for j=1:N ]

  #m.objective = Julp.AffExpr([(v[j],convert(Float64,c[j])) for j=1:N])
  #m.objective = sum([ c[j]*v[j] for j=1:N ])
  #m.objective = Julp.lpSum([ v[j]*c[j] for j=1:N ])
  m.objective = @SumExpr([c[j]*v[j] for j=1:N])

  for i = 1:M
    #lhs = sum([v[j]*A[i,j] for j=1:N])
    #lhs = Julp.AffExpr([(v[j],convert(Float64,A[i,j])) for j=1:N])
    #lhs = Julp.lpSum([ v[j] * A[i,j] for j = 1:N ])
    lhs = @SumExpr([ A[i,j]*v[j] for j = 1:N ])
    addConstraint(m, lhs <= b[i])
  end
  toc()
  println("In building the model in memory")
  tic()
  writeLP(m,"dense.lp")
  toc()
  println("Total")
end

doTest()

File: test_dense.py
# test_dense.py
# Make a big dense matrix using PuLP
from pulp import *
import random
import time

N = 1000
M = 100
A = []
for i in range(M):
  A.append([])
  for j in range(N):
    A[i].append(random.randint(1,10))
b = []
for i in range(M):
  b.append(random.randint(N,N+N*10))
c = []
for j in range(N):
  c.append(random.randint(1,10))


tic = time.time()
m = LpProblem("dense",LpMaximize)
v = LpVariable.dict("x",[j for j in range(N)],0,1)

m += lpSum(v[j] * c[j] for j in range(N))

for i in range(M):
  m += lpSum(v[j] * A[i][j] for j in range(N)) <= b[i]

m.writeLP("dense_py.lp")
toc = time.time()

print tic,toc,toc-tic
